src/RequestHandler.php

```diff
@@ -149,7 +150,7 @@ public function getNextSlave()
             $slave = array_shift($available);
 
             // slave available -> connect
-            if (!$this->tryOccupySlave($slave)) {
+            if (!$this->tryOccupySlave($slave) && !$this->redirectionTries > $this->maxRedirectionTries) {
```

Comment:
I would rather try the same solution as in the code block below:

            if (!$this->tryOccupySlave($slave)) {
                $this->loop->futureTick([$this, 'getNextSlave']);
            }

Could you see if that would work?

Reply:
/cc @dnna